tests/src/test/scala/org/apache/openwhisk/core/connector/test/EventMessageTests.scala

```diff
@@ -90,5 +90,31 @@ class EventMessageTests extends FlatSpec with Matchers {
       Activation("ns2/a", 0, toDuration(0), toDuration(0), toDuration(0), "testkind", false, 0, None, Some(42)))
   }
 
+  it should "Transform a activation with error status code" in {
+    val resultWithError =
+      """
+        |{
+        | "statusCode" : 404,
+        | "body": "Requested resource not found"
+        |}
+        |""".stripMargin
+    val a =
+      fullActivation
+        .copy(response = ActivationResponse.applicationError(resultWithError.parseJson, Some(42)))
+    Activation.from(a) shouldBe Success(
```

Comment:
Better to restrict the assertion to `statusCode` value itself instead of checking on whole assertion. Would reduce the noise